ui/src/components/home/ProfileContainerDesktop.tsx

```diff
@@ -0,0 +1,387 @@
+            ),
+          }}
+        />
+        <Typography variant="h3" fontWeight={400}>
+          New Password
+        </Typography>
+        <Divider style={{ marginTop: '2%', marginBottom: '3%' }} />
+        <TextField
+          fullWidth
+          id="outlined-adornment-password"
+          label="New Password"
+          style={{ marginBottom: '5%' }}
+          onChange={(e) => setNewPassword(e.target.value)}
+          onBlur={onVerifyPassword}
+          type={showNewPassword ? 'text' : 'password'}
+          {...props.register('newPassword')}
+          InputProps={{
+            endAdornment: (
+              <InputAdornment position="end">
+                <IconButton
+                  aria-label="toggle password visibility"
+                  onClick={handleClickShowNewPassword}
+                  onMouseDown={handleMouseDownPassword}
+                  edge="end"
+                >
+                  {showNewPassword ? <VisibilityOff /> : <Visibility />}
+                </IconButton>
+              </InputAdornment>
+            ),
+          }}
+        />
+        {/* TODO: error handling still aint right */}
+        <TextField
+          fullWidth
+          id="outlined-basic"
+          label="Confirm New Password"
+          style={{ marginBottom: '5%' }}
+          onChange={(e) => setConfirmPassword(e.target.value)}
+          onBlur={onVerifyPassword}
+          error={!matchPassword}
+          helperText={!matchPassword ? 'Password does not match!' : ' '}
+          type={showConfirmNewPassword ? 'text' : 'password'}
+          InputProps={{
+            endAdornment: (
+              <InputAdornment position="end">
+                <IconButton
+                  aria-label="toggle password visibility"
+                  onClick={handleClickShowConfirmNewPassword}
+                  onMouseDown={handleMouseDownPassword}
+                  edge="end"
+                >
+                  {showConfirmNewPassword ? <VisibilityOff /> : <Visibility />}
+                </IconButton>
+              </InputAdornment>
+            ),
+          }}
+        />
+        <Button variant="contained" type="submit" fullWidth>
+          <Typography variant="subtitle2">SAVE</Typography>
+        </Button>
+      </form>
+    </Stack>
+  );
+}
+
+const ProfileContainerDesktop: FC = (props: any): ReactElement => {
+  const user = useSelector((state: any) => state.user);
+  const { setValue, handleSubmit, register } = useForm();
+  const [userData, setUserData] = useState(null);
+  const [showEditProfile, setShowEditProfile] = useState<boolean>(false);
+  const [showChangePassword, setShowChangePassword] = useState<boolean>(false);
+  const settableDataFields = ['email', 'fullName', 'password'];
+
+  const fetchUserData = useCallback(async () => {
+    if (user.id) {
+      const { data, status } = await axios.get('/api/users/get', { params: { userId: user.id } });
+
+      if (status === 200) {
+        setUserData(data[0]);
```

Comment:
you sure the data returned is in array form? Cuz if u look at the api, it use findOne which returns an object not an array

Reply:
huh interesting, not sure why its in array. Think it has to do with the `req.params` and `req.query` in backend. @tamarayustian @yuenyauu think have to change backend to `req.query`?